packages/components/src/components/carousel/carousel.test.ts

```diff
@@ -0,0 +1,550 @@
+            <sd-carousel-item>Node 2</sd-carousel-item>
+            <sd-carousel-item>Node 3</sd-carousel-item>
+          </sd-carousel>
+        `);
+        sinon.stub(el, 'goToSlide');
+        await el.updateComplete;
+        // Act
+        const paginationItem = el.shadowRoot!.querySelectorAll('.carousel__pagination-item')[2] as HTMLElement;
+        await clickOnElement(paginationItem);
+        expect(el.goToSlide).to.have.been.calledWith(2);
+      });
+    });
+  });
+
+  describe('when `slides-per-page` attribute is provided', () => {
+    it('should show multiple slides at a given time', async () => {
+      // Arrange
+      const el = await fixture<SdCarousel>(html`
+        <sd-carousel slides-per-page="2">
+          <sd-carousel-item>Node 1</sd-carousel-item>
+          <sd-carousel-item>Node 2</sd-carousel-item>
+          <sd-carousel-item>Node 3</sd-carousel-item>
+        </sd-carousel>
+      `);
+
+      // Act
+      await el.updateComplete;
+
+      // Assert
+      expect(el.scrollContainer.style.getPropertyValue('--slides-per-page').trim()).to.be.equal('2');
+    });
+  });
+
+  describe('Navigation controls', () => {
+    describe('when the user clicks the next button', () => {
+      it('should scroll to the next slide', async () => {
+        // Arrange
+        const el = await fixture<SdCarousel>(html`
+          <sd-carousel>
+            <sd-carousel-item>Node 1</sd-carousel-item>
+            <sd-carousel-item>Node 2</sd-carousel-item>
+            <sd-carousel-item>Node 3</sd-carousel-item>
+          </sd-carousel>
+        `);
+        const nextButton: HTMLElement = el.shadowRoot!.querySelector('#carousel__navigation-button--next')!;
+        sinon.stub(el, 'next');
+        await el.updateComplete;
+        // Act
+        await clickOnElement(nextButton);
+        await el.updateComplete;
+        // Assert
+        expect(el.next).to.have.been.calledOnce;
+      });
+      describe('and carousel is positioned on the last slide', () => {
+        it('should not scroll', async () => {
+          // Arrange
+          const el = await fixture<SdCarousel>(html`
+            <sd-carousel>
+              <sd-carousel-item>Node 1</sd-carousel-item>
+              <sd-carousel-item>Node 2</sd-carousel-item>
+              <sd-carousel-item>Node 3</sd-carousel-item>
+            </sd-carousel>
+          `);
+          const nextButton: HTMLElement = el.shadowRoot!.querySelector('#carousel__navigation-button--next')!;
+          sinon.stub(el, 'next');
+          el.goToSlide(2, 'auto');
+          await oneEvent(el.scrollContainer, 'scrollend');
+          await el.updateComplete;
+          // Act
+          await clickOnElement(nextButton);
+          await el.updateComplete;
+          // Assert
+          expect(nextButton).to.have.attribute('aria-disabled', 'true');
+          expect(el.next).not.to.have.been.called;
+        });
+        // TODO: This test times out.  Seems related to be `scrollend`.
```

Comment:
pls remove code here and add to fix-ticket if its blocking from merge as ToDos might get lost.